localgov_guides.module

```diff
@@ -52,15 +53,78 @@ function localgov_guides_node_insert(NodeInterface $node) {
  */
 function localgov_guides_node_update(NodeInterface $node) {
   if ($node->bundle() == 'localgov_guides_page') {
+    if (
+      ($previous = $node->original) &&
+      ($previous->localgov_guides_parent->target_id != $node->localgov_guides_parent->target_id)
+    ) {
+      // There is a previous version of this page.
+      // The guide overview it is in isn't the same in the previous and new
+      // versions.
+      if ($old_parent = $node->original->localgov_guides_parent->entity) {
```

Comment:
I get unnerved by three level deep conditionals and loops, but you have explained it well in comments, so I can live with this.

Reply:
When writing this I had wondered about pushing all the code together into a class as we've done elsewhere. So I've done that. It doesn't really cut down the nested complexity, but gives one more function name (and injects the service to get the query).

Reply:
So https://github.com/localgovdrupal/localgov_guides/pull/27/commits/78ff650075002a00927b00555746cc139e9b0c62 maybe makes this better.

Not sure about the github automated running of the tests at the moment :-( so worth re-running them locally.